### PR TITLE
Add pkg-config files

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #32 

I reused some code here from network interfaces (the old package) and control libraries. Tested with hardware interfaces.

## Review guidelines

Estimated Time of Review: 15 minutes

### CODE REVIEW

File: source/communication_interfaces/CMakeLists.txt [hunk 1]
```diff
@@ -58,3 +71,35 @@ if(BUILD_TESTING)
...
     add_test(NAME test_${PROJECT_NAME} COMMAND test_${PROJECT_NAME})
 endif()
+
+
+# generate the version file for the config file
+write_basic_package_version_file(
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+        VERSION "${PROJECT_VERSION}"
+        COMPATIBILITY SameMajorVersion
+)
```
Comment (eeberhard) [inline source/communication_interfaces/CMakeLists.txt:81]: nitpick: inconsistent spacing of closing parenthesis

**Review by eeberhard — APPROVED** [review]
Seems good to me following the pattern we used for control libraries.

Tagging @LouisBrunner just for reference - this library is used by various hardware interfaces. It also is open source with the intention that users may write their own drivers (e.g., an interface to a specific robot or simulator). Eventually it would be nice if it can built by the CI as a static library for easier installation and included in the backend base by default.

**Comment by LouisBrunner** [comment]
> Eventually it would be nice if it can built by the CI as a static library for easier installation...

That makes sense

>  ...and included in the backend base by default.

Because you might want to build those hardware interfaces from within the backend-base? Otherwise I would imagine we make a Docker image to build the hardware interface and we just copy the interface into the backend after?

**Comment by eeberhard** [comment]
you're right, I contradicted myself a bit. It can be built and bundled with hardware targets and so doesn't necessarily need to be in the backend base.

File: source/communication_interfaces/CMakeLists.txt (hunk 1 above)
Comment (eeberhard) [reply source/communication_interfaces/CMakeLists.txt:81]: never mind, that's just how the formatting works when the arguments are listed on a new line. all good

**Pull request merged**